vm/src/obj/objbyteinner.rs

```diff
@@ -396,3 +465,23 @@ pub fn is_byte(obj: &PyObjectRef) -> Option<Vec<u8>> {
     j @ PyByteArray => Some(get_value_bytearray(&j.as_object()).to_vec()),
     _ => None)
 }
+
+pub fn is_bytes_like(obj: &PyObjectRef) -> Option<Vec<u8>> {
```

Comment:
Same note about binary predicate as below.
Also, could this return a reference instead of copying?

Reply:
not so easy...
```
Some(&objbytes::get_value(&x))                                                                                                                                                                                              
    |         ^^^^^^-----------------------^                                                                                                                                                                                              
    |         |     |                                                                                                                                                                                                                     
    |         |     temporary value created here                                                                                                                                                                                          
    |         returns a value referencing data owned by the current function 
```
this error : https://doc.rust-lang.org/error-index.html#E0515
I 'm not sure I can return anything else than a Owned value, thats to say a Vec.

Reply:
Hm, main big issue is that it's hard to mix immutable Python objects (that you can just borrow from) and mutable structures (which give you a Ref/RefMut instead of a normal reference). Similar issue exists for List/Tuple. Anyway, maybe not now, but long-term it'll probably need solving.

Reply:
As for your particular error, I think it's easy - `objbytes::get_value` already returns a reference, so I'm guessing you just don't need a `&` there.

Reply:
no it won"t
 ```
error[E0308]: mismatched types                                                                                                                                                                                                            
   --> vm/src/obj/objbyteinner.rs:489:26                                                                                                                                                                                                  
    |                                                                                                                                                                                                                                     
489 |     _i @ PyBytes => Some(objbytes::get_value(&obj)),                                                                                                                                                                                
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                  
    |                          |                                                                                                                                                                                                          
    |                          expected &[u8], found opaque type                                                                                                                                                                          
    |                          help: consider borrowing here: `&objbytes::get_value(&obj)`                                                                                                                                                
    |                                                                                                                                                                                                                                     
    = note: expected type `&[u8]`                                                                                                                                                                                                         
               found type `impl lazy_static::Deref`   
```
trying to modify objbytes::getvalue  return type to `&[u8]` gives back the previous  E0515 error.

Reply:
Anyway, I agree that it's a hard thing; it can come later, but it definitely should be done at some point.